src/dev/flang/ast/ParsedCall.java

```diff
@@ -310,15 +310,23 @@ boolean isChainedBoolRHS()
 
 
   /**
-   * Does this call a non-generic infix operator?
+   * Is this a call to an operator that may be
+   * considered valid in a chained boolean?
+   * I.e.: <,>,≤,≥,=,<=,>=,!=
    */
-  private boolean isInfixOperator()
+  private boolean isValidOperatorInChainedBoolean()
   {
     return
-      _name.startsWith("infix ") &&
-      (_actuals.size() == 1 /* normal infix operator 'a.infix + b' */                ||
-       _actuals.size() == 2 /* infix on different target 'X.Y.Z.this.infix + a b' */    ) &&
-      true; /* no check for _generics.size(), we allow infix operator to infer arbitrary number of type parameters */
+      _name.equals("infix <") ||
+      _name.equals("infix >") ||
+      _name.equals("infix ≤") ||
+      _name.equals("infix ≥") ||
+      _name.equals("infix <=") ||
+      _name.equals("infix >=") ||
+      _name.equals("infix =") ||
+      _name.equals("infix !=") ||
+      // && is used to chain the calls togehter.
+      _name.equals("infix &&");
```

Comment:
@maxteufel My understanding is that `infix ||` already works, no chaining necessary. `infix &&` is only added because this is what is used for chaining the expression together. From the documentation:
```
   *   a < b <= c < d
   *
   * which is first converted into
   *
   *   (a < {t1 := b; t1} && t1 <= c) < d
```
And currently the logic checks target as well as call itself, thus I needed to add `infix &&`.
Maybe there is room for simplification though..